### PR TITLE
feat: fix filter search contains all

### LINKED ISSUE #686: Fix way of filtering for "All" values in Earnings Table filter
This is a risky way of spotting that all values should be included.
If a user wants to filter by a grower name that starts with or contains "all", we will ignore the value and return all growers.

_Originally posted by @nmcharlton in https://github.com/Greenstand/treetracker-admin-client/pull/625#discussion_r889703220_

### PR DESCRIPTION
**Pull request by Hengxin-Liu**
## Description

**Issue(s) addressed**
If a user wants to filter by a grower name that starts with or contains "all", we will
 ignore the value and return all growers.

- Resolves #686 
**What kind of change(s) does this PR introduce?**

- [ ] Enhancement
- [x] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**

**What is the new behavior?**

## Breaking change

**Does this PR introduce a breaking change?**

### CODE REVIEW

File: src/components/EarningsTable/EarningsTable.js [hunk 1]
```diff
@@ -8,16 +8,7 @@ import {
 } from 'utilities';
 import CustomTableFilter from 'components/common/CustomTableFilter/CustomTableFilter';
 import CustomTableItemDetails from 'components/common/CustomTableItemDetails/CustomTableItemDetails';
-/**
```
Comment (nmcharlton) [inline src/components/EarningsTable/EarningsTable.js:11]: What's the reason for removing this block?

File: src/components/EarningsTable/EarningsTable.js [hunk 2]
```diff
@@ -140,7 +131,7 @@ function EarningsTable() {
     // filter out keys we don't want to submit
     Object.keys(filtersToSubmit).forEach((k) => {
       if (
-        filtersToSubmit[k] === 'all' ||
+        filtersToSubmit[k].includes('all') ||
```
Comment (nmcharlton) [inline src/components/EarningsTable/EarningsTable.js:134]: Unfortunately this causes the opposite behaviour to what we want.
User should be able to type "all" into the filter and see only results containing the string "all".
We need another way to represent _show all_ in `filtersToSubmit`.

**Review by nmcharlton — CHANGES_REQUESTED** [review]

File: src/components/EarningsTable/EarningsTable.js (hunk 1 above)
Comment (Hengxin-Liu) [reply src/components/EarningsTable/EarningsTable.js:11]: I seems like I accidentally deleted it, I fix it now

File: src/components/EarningsTable/EarningsTable.js (hunk 2 above)
Comment (nmcharlton) [reply src/components/EarningsTable/EarningsTable.js:134]: @Hengxin-Liu Are you still working on this issue?

**Pull request closed without merging**